Added clear button

Kakalanp opened the pull request:
### Clear button is finally here!

Now you can finally clear all the task you completed by the click of one button!

- Added a button in HTML.

- Styling clear button on CSS.

- Added event listener in JS to clear button.

- Modified deleteTask in JS to adapt to new functionality.

Kakalanp committed:
Added clear button

- Added event listener to clear button.

- Modified deleteTask to adapt to new functionality.

Hamzaoutdoors commented on src/index.js:
```diff
@@ -20,12 +20,32 @@ function deleteSingleTask() {
+
+    let deleteList = [];
+
+    checkboxesArr.forEach((el, i) => {
+      if (el.classList.contains('task-done')) deleteList.push(i);
+    });
```
- [x] As you can see in the screenshot below you were asked to use `Array.filter` method to clear All completed tasks from your To Do List, Kindly refactor this part to use this filter method as it's simple to use and good way for best practice.

**Syntax :** 

```JS
const newArray= arr.filter((task)=> task.completed);
```

![contactReel](https://user-images.githubusercontent.com/80895497/154805836-4595af0d-e28c-4efb-a011-18d0db9612ad.png)

Kakalanp replied:
Thanks for letting me know, but there's some kind of overlap that doesn't let me: 

Whenever I use filter, the code breaks (I suppose it has to do with the time filter takes to complete) everything after filter executes, even in the original function that called it, starts, but it doesn't end there. (Again, I assume this is because filter takes too long) The function that called the one filter is in executes again, and again, maybe 6 or 10 times, before it just doesn't.
Filter works, it does save the correct values in local storage, but everything else breaks in the process.
I'm fully available today if you could help me.

The only thing that worked is refreshing the page after clear, but I don't know if that should be done

Hamzaoutdoors reviewed (changes requested):
Hey @Kakalanp, I am **Hamza** your reviewer for this PR 😸 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![image](https://media.giphy.com/media/42AFvimyQTNQW44PNG/giphy.gif)

## To Highlight
✔️ The repo looks professional
✔️ Git/GitHub flow best practices have been followed
✔️ There are no linter errors
✔️ You have a descriptive README

## Required Changes ♻️
- [x]  Kindly add instruction on installing dependencies(`npm install` or `npm i`) and running your project locally(`npm start`)

_See the inline comments below for more required changes 😃_  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
**N.A**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @hamzaoutdoors in your question so I can receive the notification ^_^

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Kakalanp committed:
Update README.md

It now contains steps to install dependencies

mke2111 commented:
Project Under Second Opinion Review.
Kindly wait for a response. 😄

mke2111 reviewed (commented):
Hello @Kakalanp  👋,

_This is an additional review as a result of the second opinion request._

Great work with this app so far. 👍🏽 

Using the filter method is a requirement for this project. Since you mentioned you were getting some errors once you applied this method, I suggest you schedule a call with a SCR ASAP. I reached out to you to see how best we can solve this together, but I guess you were not available. Hoping to see you soon to get this fixed.

In meantime, your project is not yet ready for approval.

Keep rocking 🚀!

Kakalanp committed:
deleteTask now uses filter

- changed the clear button to reload the page after changes are made, this way filter can be used in this scenario.

Kakalanp committed:
Merge branch 'Clear-tasks' of https://github.com/Kakalanp/TO-DO-list into Clear-tasks

Kakalanp commented:
### To my next code reviewer:

- I'll appreciate every suggestion you can give me, but I must ask you to have consideration over the limited time frame I have left and the effort I'm putting to make this project get approved. My last change includes the filter method, and I really hope this can get the project approved.

- Thank you very much for taking the time to read this petition, I really hope it helps you have some perspective on how desperate and frustrated I am by this at the time being.

Gambit142 commented on src/index.js:
```diff
@@ -20,12 +20,33 @@ function deleteSingleTask() {
+
+    let deleteList = [];
+
+    checkboxesArr.forEach((el, i) => {
+      if (el.classList.contains('task-done')) deleteList.push(i);
+    });
```
- [x] Hi, you did not implement the filter function here. It would be nice to meet specific project requirements because, in the real world, you might be given such tasks and based to implement them using the company's standard code. 

_**Hint**_: What you want to do is to filter through the array and return those tasks that do not have their `completed` key set to `true`. Kindly refactor the code to use `array.filter` as it is a requirement for this milestone 👍 

### You can reach out to **_RoyM_** on slack and schedule a call with him. He is a senior code reviewer and he has asked you to schedule a call with him. 🙏

Kakalanp replied:
Thanks for your feedback, but I mush inform you that I did: https://github.com/Kakalanp/TO-DO-list/blob/c712489940d80e8d01b2587ed59b3e9931671d72/src/modules/deleteTask.js#L9
It's not the same way you're suggesting, but I did implement the changes.

Gambit142 reviewed (changes requested):
Hi @Kakalanp 👋, 

### I understand what you are going through, but implementing project requirements is critical to succeeding in the development world. 

 ## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

- [ ] On line 1 of your `index.js file`, you used this piece of code `/* eslint-disable no-unused-vars */`. It would be nice to remove it and fix the linters error within your code. The reason for the unused var is on line 45 where you have this piece of code `const newTask = new Task(addTask.value);`. The variable `newTask` is not used again. 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Davidosky007 reviewed (commented):
Hello @Kakalanp  👋,

_This is an additional review as a result of the second opinion request._

Hi, Great job working so hard on the requested changes kindly submit another request on your dashboard so that the next reviewer would know to approve 👍 

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away :)

Keep rocking 🚀!

Lordkaito reviewed (approved):
Hi @Kakalanp, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
